### PR TITLE
[WIP] OMG containers

### PR DESCRIPTION
**Pull request by substars**
This is my experimental Dockerization branch. It's a WIP, don't merge!

what is done:
- [x] generate front end assets
- [x] spin up (most) containers
- [x] run db migrations 
- [x] tweak config files etc. to be produce
- [x] start Rails app and all the services it needs
- [x] serve frontend assets

work left to do:
- [x] `rake db:seed` fails with a strange error
- [x] make sure data import works
- [x] get Node server running
- [ ] HTTPS 😓 
- [ ] clean up docs: consolidate READMES etc. 
- [ ] clean up commits so they make sense
- [ ] make containers production-ready (users, data volumes, security, RAILS_ENV, ???)

/cc @wbushey

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM phusion/baseimage:0.9.18 
+FROM phusion/passenger-ruby19:0.9.18
```
Comment (garadox) [inline Dockerfile:1]: Am I right in thinking this Dockerfile represents the `railsweb` container? Just making sure.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,12 +1,56 @@
...
+      - "9200:9200"
+      - "9300:9300"
   db:
-    image: postgres
-  app:
+    image: mdillon/postgis
+    ports:
+      - "5432:5432"
+  railsweb:
     build: .
```
Comment (garadox) [inline docker-compose.yml:24]: I think you can use the Dockerfile to build this railsweb container, then reference it by name here. You would run something like `docker build -t railsweb .` to build the above container, then update this reference in docker-compose.yml to use the `railsweb` image. 

That should remove a lot of needed dependencies on the client machine itself, as the first container will contain them all.

**Comment by substars** [comment]
Thanks @garadox, yes the Dockerfile is for railsweb, and I just added a different Dockerfile for the node API app. We will probably want to at least put dev/compile-time dependencies into a third Dockerfile, will definitely look into building pre-baked images too.

**Comment by substars** [comment]
see #11 - that contains all this work and more

**Pull request closed without merging**